From 05e7c15997d29db0d251335c61037e23b481098c Mon Sep 17 00:00:00 2001
From: Pino Toscano <ptoscano@redhat.com>
Date: Fri, 5 Dec 2014 15:23:28 +0100
Subject: [PATCH 079/102] sysprep: user-account: mark "changed file" if users
 were removed

Removing an user causes /etc/passwd, /etc/shadow, and /etc/group to
change, so mark the side effects as such if any user has been removed.

(cherry picked from commit d4ffaad98390141f1830c1aa41425e79c23ecb02)
---
 sysprep/sysprep_operation_user_account.ml | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/sysprep/sysprep_operation_user_account.ml b/sysprep/sysprep_operation_user_account.ml
index fc39bc8..b44fbfb 100644
--- a/sysprep/sysprep_operation_user_account.ml
+++ b/sysprep/sysprep_operation_user_account.ml
@@ -27,6 +27,7 @@ module G = Guestfs
 
 let user_account_perform g root =
   let typ = g#inspect_get_type root in
+  let changed = ref false in
   if typ <> "windows" then (
     g#aug_init "/" 0;
     let uid_min = g#aug_get "/files/etc/login.defs/UID_MIN" in
@@ -40,6 +41,7 @@ let user_account_perform g root =
         let uid = g#aug_get uid in
         let uid = int_of_string uid in
         if uid >= uid_min && uid <= uid_max then (
+          changed := true;
           g#aug_rm userpath;
           let username =
             let i = String.rindex userpath '/' in
@@ -54,7 +56,7 @@ let user_account_perform g root =
         )
     ) users;
     g#aug_save ();
-    []
+    if !changed then [ `Created_files ] else []
   )
   else []
 
-- 
1.9.3

